### PR TITLE
PARQUET-849: Upgrade to Thrift 0.10 in thirdparty toolchain, build Thrift in Travis CI on OS X

### PR DESCRIPTION
**Pull request by wesm**
Closes #251.

Also resolves PARQUET-871 by adding `-DPARQUET_CXXFLAGS=...` option

### CODE REVIEW

File: cmake_modules/ThirdpartyToolchain.cmake [hunk 1]
```diff
@@ -42,41 +42,42 @@ set(EP_C_FLAGS "${CMAKE_C_FLAGS} ${CMAKE_C_FLAGS_${UPPERCASE_BUILD_TYPE}} -fPIC"
...
+  ELSE()
+    set(THRIFT_STATIC_LIB "${THRIFT_PREFIX}/lib/libthrift.a")
+  ENDIF()
   set(THRIFT_COMPILER "${THRIFT_PREFIX}/bin/thrift")
   set(THRIFT_VENDORED 1)
-  set(THRIFT_CONFIGURE_COMMAND
-      ./configure "CFLAGS=${EP_C_FLAGS}" "CXXFLAGS=${EP_CXX_FLAGS}" --without-qt4 --without-c_glib --without-csharp --without-java --without-erlang --without-nodejs --without-lua --without-python --without-perl --without-php --without-php_extension --without-ruby --without-haskell --without-go --without-d --with-cpp "--prefix=${THRIFT_PREFIX}")
+  set(THRIFT_CMAKE_ARGS -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+                        "-DCMAKE_CXX_FLAGS=${EP_CXX_FLAGS}"
+                        "-DCMAKE_C_FLAGS=${EX_C_FLAGS}"
```
Comment (xhochy) [inline cmake_modules/ThirdpartyToolchain.cmake]: Typo, this should be `EP_C_FLAGS`

Comment (wesm) [inline cmake_modules/ThirdpartyToolchain.cmake]: done + rebase

**Comment by wesm** [comment]
here's the build on my fork, just waiting for OS X to run: https://travis-ci.org/wesm/parquet-cpp/builds/203858538

**Comment by jlafaye** [comment]
Thank you for fnishing the job. No time to work on it during work hours today.

**Comment by xhochy** [comment]
Build fails due to passing `-Werror` to Thrift. @wesm feel free to do any solution you like on that problem (https://issues.apache.org/jira/browse/PARQUET-871), I'm happy as long as I have an easy way to pass options to all dependencies.

**Comment by wesm** [comment]
Done

**Comment by wesm** [comment]
The build passed on my fork, so I am merging: https://travis-ci.org/wesm/parquet-cpp/builds/203924826. Thank you!

**Comment by wesm** [comment]
@jlafaye if you can create an account on issues.apache.org I will assign PARQUET-849 to you

**Comment by jlafaye** [comment]
Account created, userid is 'jlafaye'

**Pull request closed without merging**